pkg/providers/snow/entry.go

```diff
@@ -0,0 +1,78 @@
+package snow
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/aws"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/networkutils"
+	providerValidator "github.com/aws/eks-anywhere/pkg/providers/validator"
+)
+
+type ConfigManagerEntry struct {
+	configManager *cluster.ConfigManager
+	validator     *Validator
+	defaulters    *Defaulters
+}
+
+func NewConfigManagerEntry(aws aws.Clients, writer filewriter.FileWriter) *ConfigManagerEntry {
+	configManager := cluster.NewConfigManager()
+	return &ConfigManagerEntry{
+		configManager: configManager,
+		validator:     NewValidator(aws),
+		defaulters:    NewDefaulters(aws, writer),
+	}
+}
+
+func (cm *ConfigManagerEntry) SetDefaultsAndValidate(ctx context.Context, config *cluster.Config) error {
+	if err := cm.configManager.Register(cm.snowEntry(ctx)); err != nil {
```

Comment:
I would recommend creating the `configManager` here on the fly. Register will store the entry in the config manager, hence if you run this twice, it will run the old entry and the new entry